pkg/controller/postgres/deletion_protection.go

```diff
@@ -83,32 +84,55 @@ func getRequeueTime(ctx context.Context, inst client.Object, deletionTime *metav
 }
 
 func getPatchObjectFinalizer(log logr.Logger, inst client.Object, op jsonpatch.JSONop) (client.Patch, error) {
-	if op == jsonpatch.JSONopNone {
-		return nil, nil
-	}
-
 	// handle the case if crossplane or something else decides to add more finalizers, or if
 	// the finalizer is already there.
 	index := len(inst.GetFinalizers())
+	dupes := []int{}
 	for i, finalizer := range inst.GetFinalizers() {
 		if finalizer == finalizerName {
 			index = i
+			dupes = append(dupes, i)
 		}
 	}
 
+	// if this is a noop and we have exactly one or no dupe, then we're done here
+	if op == jsonpatch.JSONopNone && (len(dupes) == 1 || len(dupes) == 0) {
```

Comment:
WHy do you expect to have exactly 2 finalizers before proceeding?

Reply:
I thought you already had such check before but written differently, let's hope this works.

Reply:
The previous logic didn't really act on the amount of finalizers found apart from knowing which one to remove.

This here is basically the main check to see if we have multiple finalizers or not, if we do, then the rest of this function will determine that and clean them up.

So far all of the unit tests are succeeding and it worked on the lab.